api/envoy/config/core/v3/base.proto

```diff
@@ -313,6 +313,23 @@ message HeaderMap {
   repeated HeaderValue headers = 1;
 }
 
+// A file that is watched for changes, e.g. by inotify on Linux.
+message WatchedFile {
```

Comment:
Should we use this in the filesystem runtime code and deprecate whatever is there now? This seems complicated enough that it would be good to have it in one place? This assumes that the functionality is basically the same.

Reply:
This is a watch on a file rather than a tree, the filesystem runtime has a bunch of stuff around overlays as well, not sure how much common ground is left..